debug/mutex.py

```diff
@@ -0,0 +1,57 @@
+import gdb
+import sys
+import os
+sys.path.append(os.path.join(os.getcwd(), 'debug'))
+import tailq
+
+
+def get_mutex_owner(mtx):
+    state = mtx['mtx_state']
+    td_pointer = gdb.lookup_type('struct thread').pointer()
+    owner = state.cast(td_pointer)
+    if owner:
+        owner = owner.dereference()
+        return owner['td_name'].string()
+    return None
+
+
+def get_threads_blocked_on_mutex(mtx):
+    ts = mtx['turnstile']
+    return map(lambda t: t['td_name'].string(), tailq.collect_values(ts['td_queue'], 'td_lock'))
+
+
+def pretty_list_of_string(lst):
+    res = "["
+    for i in range(0, len(lst)):
+        res = res + str(lst[i])
+        if(i != len(lst) - 1):
+            res = res + ", "
+    res = res + "]"
+    return res
```

Comment:
I had to change string representation of list slightly because doing `print lst` printed the list like this:

`[u't1', u't2', u't3']`

And I wanted the list to be printed like this:

`[t1, t2, t3]`

Same goes for `repr(lst)`. I totally didn't think of functional way to create this function. I like @dyniec's solution the most since it's the most readable to me.